### PR TITLE
Transmission parsing v2

### PR DESCRIPTION
**Pull request by bmagyar**

### CODE REVIEW

File: hardware_interface/src/component_parser.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #include <unordered_map>
 #include <vector>
 #include <regex>
+#include <iostream>
```
Comment (bmagyar) [inline hardware_interface/src/component_parser.cpp:21]:
```suggestion
```

File: hardware_interface/src/component_parser.cpp [hunk 2]
```diff
@@ -107,6 +112,38 @@ std::string get_attribute_value(
...
+  while (params_it) {
+    // Fill the map with parameters
+    const auto tag_name = params_it->Name();
+    if (strcmp(tag_name, parameter_name) == 0) {
+      const auto tag_text = params_it->GetText();
+      if (!tag_text) {
+        throw std::runtime_error("text not specified in the " + std::string(tag_name) + " tag");
+      }
+      std::cerr << "SUCCESSSSSSSSSSSss " << tag_name << " and param name was " << parameter_name <<
+        " and value was " << tag_text << std::endl;
```
Comment (bmagyar) [inline hardware_interface/src/component_parser.cpp:138]:
```suggestion
```

**Review by destogl — APPROVED** [review]
This seems legit!

**Review by jordan-palacios — APPROVED** [review]
Looks perfect.

**Pull request merged**